### PR TITLE
Middlewares

### PR DESCRIPTION
**Pull request by shernanm**
## Summary

* The use of middleware is implemented in the endpoints of users and albums.

## Trello Card

https://trello.com/c/3JYipc49

### CODE REVIEW

File: app/controllers/album.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const albumsService = require('../services/albums');
 const logger = require('../logger');
 const message = require('../constants');
-const errors = require('../errors');
+// const errors = require('../errors');
```
Comment (raparicio6) [inline app/controllers/album.js]: dont leave comments

File: app/controllers/user.js [hunk 2]
```diff
@@ -1,21 +1,18 @@
...
     .then(userService.userRegister)
     .then(response => {
-      logger.info(`User ${response.firstName} ${response.lastName} created successfully`);
+      logger.info(`User ${user.firstName} ${user.lastName} created successfully`);
       return res.status(201).send({
-        firstName: response.firstName,
-        lastName: response.lastName
+        firstName: user.firstName,
+        lastName: user.lastName,
+        message: `User has been created successfully with ID ${response.id}`
```
Comment (raparicio6) [inline app/controllers/user.js]: Why do you send a message in the response? You dont do this in APIs Rest, you just send the data with a status code

File: app/schemas/user.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+const { User } = require('../models');
```
Comment (raparicio6) [inline app/schemas/user.js:1]: great work!

File: app/util.js [hunk 4]
```diff
@@ -1,21 +1,14 @@
...
-exports.paramsValidation = (pass, email) => {
-  const regexPass = /^([a-z0-9]){8,}$/i;
-  const regexMail = /^[a-z0-9._-]+@wolox.(co|cl|com|ar|com.ar)+$/i;
-  return !regexPass.test(pass) || !regexMail.test(email);
-};
 
 exports.passwordEncryption = user =>
   bcrypt
-    .hash(user.password, session.salt_rounds)
+    .hash(user.password, 10)
```
Comment (raparicio6) [inline app/util.js]: dont harcode this

**Review by raparicio6 — CHANGES_REQUESTED** [review]

File: app/controllers/user.js [hunk 5]
```diff
@@ -1,18 +1,14 @@
...
 const userService = require('../services/users');
-const errors = require('../errors');
 const logger = require('../logger');
 
 exports.createUser = (req, res, next) => {
-  const { password, email } = req.body;
-  if (validations.paramsValidation(password, email)) {
-    return next(errors.invalidParameters('The email or password does not meet the conditions.'));
-  }
+  const user = req.body;
```
Comment (molinajulian) [inline app/controllers/user.js]: You can deconstruct this in req . `({ body }, res, next)`

File: app/middlewares/checkSchema.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const { checkSchema, validationResult } = require('express-validator/check');
+
+const error = require('../errors');
+
+const checkValidationResult = (req, res, next) => {
+  const errors = validationResult(req);
+  return errors.isEmpty() ? next() : res.status(400).json({ errors: errors.array(), ...error.badRequest() });
```
Comment (molinajulian) [inline app/middlewares/checkSchema.js]: You don't need set status. Check `errors.js` and add key `BAD_REQUEST` with status code `400`. Also, `json` is not necessary.  If you want send a json, is more easy, use `.send({ key: value})`.

File: app/util.js [hunk 7]
```diff
@@ -3,19 +3,13 @@ const bcrypt = require('bcrypt');
...
   bcrypt
     .hash(user.password, session.salt_rounds)
     .then(password => ({
       ...user,
       password
     }))
-    .catch(err => {
-      throw errors.encryptionError(`Error trying to encrypt the password. ${err.message}`);
+    .catch(() => {
+      throw errors.encryptionError();
```
Comment (molinajulian) [inline app/util.js]: Perfect! Maybe you can obtain `message` and pass to `encryptionError` and add it to default message for this error.

File: test/users.test.js [hunk 8]
```diff
@@ -45,14 +47,15 @@ describe('POST /users', () => {
...
           .then(response => {
-            const { message, internal_code } = response.body;
-            expect({ status: response.statusCode, message, internal_code }).toStrictEqual({
-              status: 503,
-              message: 'Error processing request in database.',
-              internal_code: 'database_error'
+            const { message, internalCode } = response.body;
+            expect({ status: response.statusCode, message, internalCode }).toStrictEqual({
+              status: 400,
+              message: 'Invalid request parameters',
```
Comment (molinajulian) [inline test/users.test.js]: This is not the same with what we talked about. You must be send an array for errors in message :( . Check this in all test

**Review by molinajulian — CHANGES_REQUESTED** [review]

File: app/controllers/user.js [hunk 9]
```diff
@@ -1,22 +1,16 @@
...
 const logger = require('../logger');
 
-exports.createUser = (req, res, next) => {
-  const { password, email } = req.body;
-  if (validations.paramsValidation(password, email)) {
-    return next(errors.invalidParameters('The email or password does not meet the conditions.'));
-  }
-  return validations
-    .passwordEncryption(req.body)
+exports.createUser = ({ body }, res, next) =>
```
Comment (raparicio6) [inline app/controllers/user.js:5]: for convention we always put `req` here

**Review by raparicio6 — CHANGES_REQUESTED** [review]

File: app/controllers/user.js (hunk 9 above)
Comment (shernanm) [reply app/controllers/user.js:5]: I did it at the suggestion of Moli

Comment (raparicio6) [reply app/controllers/user.js:5]: ok its fine in that case

**Review by molinajulian — APPROVED** [review]

**Review by raparicio6 — APPROVED** [review]

**Pull request merged**